lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+
+This function should go inside the `App` component, after `studentData` is defined, and before the `return` statement.
+
+<!-- prettier-ignore-start -->
+```js
+  const updateStudentData = updatedStudent => {
+    const students = studentData.map(student => {
+      if (student.id === updatedStudent.id) {
+        return updatedStudent;
+      } else {
+        return student;
+      }
+    });
+
+    setStudentData(students);
+  }
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:300px;"> Piece of Code </div> | Notes                                                                                                                                                                |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const updateStudentData = `                        | We're creating a function named `updateStudentData`.                                                                                                                 |
+| `updatedStudent`                                    | This function accepts one argument: an object that holds updated student data. We expect this object to have the key `id`, at the very least.                        |
+| `const students = ;`                                | For this function, we'll create a helper array, `students` to contain the updated student data.                                                                      |
+| `studentData.map(...)`                              | We'll map over `studentData`...                                                                                                                                      |
+| `student => {...}`                                 | Inside `map`, we'll reference each item with `student`...                                                                                                            |
+| `if (student.id === updatedStudent.id) {`           | If we found the student that our `updatedStudent` is updating...                                                                                                     |
+| `return updatedStudent`                             | The item we should map to our `students` array is the updated student data!                                                                                          |
+| `else { return student; }`                          | Otherwise, we should map the unchanged `student` into `students`.                                                                                                    |
+| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` on state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
```

Comment:
there are a few places where the phrasing "on state" is used. Was this left over from the idea that there would be an actual state object (class-like React) and these values would be properties "on" that state object?
```suggestion
| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` in our state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
```